src/bitops.c

```diff
@@ -429,7 +429,8 @@ int getBitOffsetFromArgument(client *c, robj *o, uint64_t *offset, int hash, int
     if (usehash) loffset *= bits;
 
     /* Limit offset to server.proto_max_bulk_len (512MB in bytes by default) */
-    if ((loffset < 0) || (loffset >> 3) >= server.proto_max_bulk_len)
+    if (!(c->flags & CLIENT_MASTER) &&
+        ((loffset < 0) || (loffset >> 3) >= server.proto_max_bulk_len))
```

Comment:
i suppose we wanna keep the check for negative offset even for master client.
we don't really expect that, but no reason to allow it anyway..
```suggestion
    if (loffset < 0 || (!(c->flags & CLIENT_MASTER) && (loffset >> 3) >= server.proto_max_bulk_len))
```